h2o-r/h2o-package/R/frame.R

```diff
@@ -3782,6 +3782,67 @@ h2o.arrange <- function(x, ...) {
   .newExpr("sort", x, by-1L, as.numeric(ascend))
 }
 
+generateColInd <-function(data, by) {
+  ### handle the columns
+  # we accept: c('col1', 'col2'), 1:2, c(1,2) as column names.
+  if(base::is.character(by)) {
+    group.cols <- match(by, colnames(data))
+    if (any(is.na(group.cols)))
+    stop('No column named ', by, ' in ', substitute(data), '.')
+  } else if(is.integer(by)) {
+    group.cols <- by
+  } else if(is.numeric(by)) {   # this will happen eg c(1,2,3)
+    group.cols <- as.integer(by)
+  }
+
+  if(group.cols <= 0L || group.cols > ncol(data)) {
+   stop('Column ', group.cols, ' out of range for frame columns ', ncol(data), '.')
+  }
+
+  if (anyDuplicated(by)) stop("Some duplicate column names have been provided")
+
+  return(group.cols)
+}
+
+#' First, sorts an H2OFrame by columns in sortby and sort directions specified.
+#' Next, groupby groups are formed by the columns specified in by.
```

Comment:
"Next, groupby groups are formed by the columns specified in by." 
Should that be:
"Next, groupby groups are formed by the columns specified in groupbyCols" instead? (I don't see "by" as a parameter.)

Reply:
Angela:  Let me redo this and make it better for you.   Stay tuned....